.github/workflows/ci.yml

```diff
@@ -11,9 +11,26 @@ permissions:
   contents: read
 
 jobs:
-  nix-fmt:
+  # We probably should replace with a different faster CI soon,
+  # but atleast this gives us confidence for now
+  nix-build:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
     - uses: cachix/install-nix-action@v27
-    - run: nix fmt -- --fail-on-change
+    - uses: DeterminateSystems/magic-nix-cache-action@v2
+    - run: nix run github:Mic92/nix-fast-build -- --no-nom
+  # all builds combined consume too much disk space... we should soon switch to something else
+  nixos:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # we actually cannot build umbriel just now
+        # but it also looks like it doesn't run important stuff
+        #machine: [caliban, umbriel]
+        machine: [caliban]
```

Comment:
https://github.com/marketplace/actions/maximize-build-disk-space can give you another 7-10GB of disk space. Maybe that's enough?